doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+
+#### Usage
+
+```lua
+focusable = true,
+```
+
+---
+
+### `zindex`
+
+- **Type:** `number`
+- **Description:**
+  - Sets the order of the popup on z-axis
+  - Popup with higher the `zindex` goes on top of popups with lower `zindex`
+
+#### Usage
+
+```lua
+zindex = 20,
+```
+
+---
+
+### `buf_options`
+
+- **Type:** `table`
+- **Description:** Contains all buffer related options
+
+#### Usage
+
+```lua
+buf_options = {
+    modifiable = false,
+    readonly = true,
+},
+```
+
+#### Descriptions
+
+| Value        | Description                                                                  |
+| ------------ | ---------------------------------------------------------------------------- |
+| `modifiable` | If `false` the buffer contents cannot be changed. <br> Check `:h modifiable` |
+| `readonly`   | If `true`, writes fail unless you use a `!` <br> Check `:h readonly`         |
+
+---
+
+### `win_options`
+
+- **Type:** `table`
+- **Description:** Contains all window related options
+
+#### Usage
+
+```lua
+win_options = {
+    winblend = 10,
+    winhighlight = "Normal:Normal",
+},
+```
+
+#### Descriptions
+
+| Value          | Description                                                                               |
+| -------------- | ----------------------------------------------------------------------------------------- |
+| `winblend`     | Transparency for a floating window. <br> Goes from `0` to `100`. <br> Check `:h winblend` |
+| `winhighlight` | Window local highlights. <br> Check `:h winhighlight`                                     |
+
+## Method Descriptions
+
+### `popup:mount()`
+
+- Mounts the popup
+
+```lua
+popup:mount()
+```
+
+---
+
+### `popup:unmount()`
+
+- Unmounts the popup
+
+```lua
+popup:unmount()
+```
+
+---
+
+### `popup:map(mode, key, handler, opts, force)`
+
+- Sets keymap for this popup.
+- If keymap was already set and `force` is not `true` it returns `false`, otherwise returns `true`
+
+```lua
+local ok = popup:map("n", "<esc>", function(bufnr)
+    print("ESC pressed in Normal mode!")
+end, { noremap = true })
+```
+
+#### Descriptions
+
+| Value     | Description                           |
+| --------- | ------------------------------------- |
+| `mode`    | Neovim `mode` <br> See `:h vim-modes` |
+| `key`     | Trigger for the map                   |
+| `handler` | Function to fire on that key trigger  |
+| `opts`    | TODO: what goes here                  |
+| `force`   | TODO: what goes here                  |
+
+---
+
+### `popup:on(event, handler, options)`
+
+- Defines `autocmd` to run on specific events for this popup
+
+```lua
+local event = require("nui.utils.autocmd").event
+
+popup:on({ event.BufLeave }, function()
+    popup:unmount()
+end, { once = true })
+```
+
+#### Descriptions
+
+| Value     | Description                         |
+| --------- | ----------------------------------- |
+| `event`   | Neovim `event` <br> See `:h events` |
```

Comment:
This can be a list of event names or just a single event name, or even comma separated event names. So all this works:

```lua
{ event.BufLeave, event.BufDelete }
-- or
{ event.BufLeave, "BufDelete" }
-- or
event.BufLeave
-- or
"BufLeave"
-- or
"BufLeave,BufDelete"
```